### PR TITLE
Rewrite imaginary argument to acos as asin

### LINKED ISSUE #17141: simplify(x**(1 / acos(I))) recursion error
```
>>> from sympy import *
>>> x = Symbol('x')
>>> e = x**(1 / acos(I))
>>> print(simplify(e))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/e/se/sympy/simplify/simplify.py", line 559, in simplify
    expr = bottom_up(expr, lambda w: getattr(w, 'normal', lambda: w)())
  File "/home/e/se/sympy/simplify/simplify.py", line 1077, in bottom_up
    rv = F(rv)
  File "/home/e/se/sympy/simplify/simplify.py", line 559, in <lambda>
    expr = bottom_up(expr, lambda w: getattr(w, 'normal', lambda: w)())
  File "/home/e/se/sympy/core/expr.py", line 2141, in normal
    n, d = self.as_numer_denom()
  File "/home/e/se/sympy/core/power.py", line 1322, in as_numer_denom
    return self.func(n, exp), self.func(d, exp)
  File "/home/e/se/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/e/se/sympy/core/power.py", line 283, in __new__
    if abs(e).is_infinite:
  File "/home/e/se/sympy/core/expr.py", line 167, in __abs__
    return Abs(self)
  File "/home/e/se/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/e/se/sympy/core/function.py", line 462, in __new__
    result = super(Function, cls).__new__(cls, *args, **options)
  File "/home/e/se/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/e/se/sympy/core/function.py", line 277, in __new__
    evaluated = cls.eval(*args)
  File "/home/e/se/sympy/functions/elementary/complexes.py", line 510, in eval
    return exp(re(exponent*z))
  File "/home/e/se/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/e/se/sympy/core/function.py", line 462, in __new__
    result = super(Function, cls).__new__(cls, *args, **options)
  File "/home/e/se/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/e/se/sympy/core/function.py", line 277, in __new__
    evaluated = cls.eval(*args)
  File "/home/e/se/sympy/functions/elementary/complexes.py", line 82, in eval
    real_imag = term.as_real_imag(ignore=arg)
  File "/home/e/se/sympy/core/mul.py", line 783, in as_real_imag
    r, i = a.as_real_imag()
  File "/home/e/se/sympy/functions/elementary/exponential.py", line 693, in as_real_imag
    abs = Abs(self.args[0].expand(deep, **hints))
  File "/home/e/se/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/e/se/sympy/core/function.py", line 462, in __new__
    result = super(Function, cls).__new__(cls, *args, **options)
  File "/home/e/se/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/e/se/sympy/core/function.py", line 277, in __new__
    evaluated = cls.eval(*args)
  File "/home/e/se/sympy/functions/elementary/complexes.py", line 510, in eval
    return exp(re(exponent*z))
  File "/home/e/se/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/e/se/sympy/core/function.py", line 462, in __new__
    result = super(Function, cls).__new__(cls, *args, **options)
  File "/home/e/se/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/e/se/sympy/core/function.py", line 277, in __new__
    evaluated = cls.eval(*args)

...

    retval = cfunc(*args, **kwargs)
RecursionError: maximum recursion depth exceeded while calling a Python object
```

### PR DESCRIPTION
**Pull request by oscargus**
#### References to other Issues or PRs

Partly fixes #17141 

#### Brief description of what is fixed or changed
`acos` with a purely imaginary input argument is rewritten to `asin`. This avoids the recursion error as happened in #17141.

#### Other comments

I also improved the documentation of some trigonometric functions a bit.

#### Release Notes

* functions
    * `acos` with purely imaginary input is automatically rewritten to `asin`.

### CODE REVIEW

File: sympy/functions/elementary/trigonometric.py [hunk 1]
```diff
@@ -2616,10 +2616,10 @@ class asec(InverseTrigonometricFunction):
...
     .. math::
-        \lim_{z \to 0}-i\frac{\log\left(-\sqrt{1 - z^2} + 1}{z}
+        \lim_{z \to 0}-i\frac{\log\left(-\sqrt{1 - z^2} + 1\right)}{z}
 
-    simplifies to :math:`-i*\log(z/2 + O(z^3))` which ultimately evaluates to
-    ``zoo``.
+    simplifies to :math:`-i\log\left(z/2 + O\left(z^3\right)\right)` which
+    ultimately evaluates to ``zoo``.
 
     As ``asex(x)`` = ``asec(1/x)``, a similar argument can be given for
```
Comment (jksuom) [inline sympy/functions/elementary/trigonometric.py]: asex?

Comment (oscargus) [reply sympy/functions/elementary/trigonometric.py]: Well spotted. Will change it! (The trig documentation could need a major overhaul...)

File: sympy/simplify/tests/test_simplify.py [hunk 2]
```diff
@@ -820,3 +820,19 @@ def test_issue_7971():
...
+def test_issue_17141():
+    # Check that there is no RecursionError
+    assert simplify(x**(1 / acos(I))) == x**(2/(pi - 2*I*log(1 + sqrt(2))))
+    assert simplify(acos(-I)**2*acos(I)**2) == \
+           log(1 + sqrt(2))**4 + pi**2*log(1 + sqrt(2))**2/2 + pi**4/16
+    assert simplify(2**acos(I)**2) == 2**((pi - 2*I*log(1 + sqrt(2)))**2/4)
+
+    # However, for a complex number it still happens
+    if PY3:
+        raises(RecursionError, lambda: simplify(2**acos(I+1)**2))
```
Comment (smichr) [inline sympy/simplify/tests/test_simplify.py]: You can detect numbers with `I` in them as:
```python
>>> from sympy.core.evalf import pure_complex as f
>>> f(I)
(0, 1)
>>> f(1+2*I)
(1, 2)
>>> f(3) is None
True
```

Comment (oscargus) [reply sympy/simplify/tests/test_simplify.py]: I think it boils down to the question in #17155 so I am hoping to get some feedback there.

**Pull request merged**